docs/tutorials/alpa_on_slurm.rst

```diff
@@ -0,0 +1,320 @@
+
+Please make sure you have or can create an environment with required dependencies installed for Alpa.
+These dependencies are described in `Install Alpa <https://alpa.ai/install.html>`_.
+
+    .. note::
+
+        One easy way to create an environment with all dependencies installed is to create an environment using package management systems like `conda <https://docs.conda.io/en/latest/>`_ and setup the environment running an interactive job on Slurm.
+        The interact job on Slurm can allow you to run and experiment in the exact way you do locally.
+        To run a interactive job on Slurm, use the command :code:`interact`.
+
+Create :code:`sbatch` Script
+****************************
+
+Usually large jobs like Alpa is run through sbatch on Slurm using a :code:`sbatch` script. :code:`sbatch` scripts are bash scripts with :code:`sbatch` options specified using :code:`#SBATCH <options>`.
+The Slurm cluster takes sbatch scripts submitted using command sbatch then queue the job specified by the script for execution.
+When Slurm executes the script, the script works exactly the same as a shell script.
+A :code:`sbatch` script to run Alpa can be roughly summarized as four parts: resources setup, load dependencies, Ray startup, and run Alpa.
+
+The first step is to create a :code:`sbatch` script in your directory, usually named as a :code:`.sh` file.
+Here, this guide asumes the script is included in a file :code:`run_alpa_on_slurm.sh`.
+Just like a shell script, the :code:`sbatch` script starts with a line specifying the path to interpreter:
+
+    .. code:: bash
+
+        #!/bin/bash
+
+Resources Setup
+================
+
+The first lines in your sbatch script is used to specify resources for the job like all the options you specify when running srun or sbatch.
+These usually includes the name of the job, partition the job should go to, CPU per task, memory per CPU, number of nodes, number of tasks per node, and time limit for the job.
+
+    .. code:: bash
+
+        #SBATCH --job-name=textgen_multinode_test
+        #SBATCH --partition=GPU
+        #SBATCH --nodes=2
+        #SBATCH --tasks-per-node=1
+        #SBATCH --cpus-per-task=1
+        #SBATCH --mem-per-cpu=1GB
+        #SBATCH --gpus-per-node=v100-16:8
+        #SBATCH --time=00:10:00
+
+    .. note::
+
+        Setting the resources needed in the sbatch script is equivalent to setting them submitting the job to Slurm running :code:`sbatch <options> <sbatch script>`.
+
+Load Dependencies
+=================
+
+The next step is to setup the environment with Alpa's dependencies installed.
+In some Slurm clusters, CUDA, NVCC, and CuDNN are packed in containers that can be loaded directly. Here, we provide an example that loads a combination of available container and user-defined environment from package management systems.
+To load directly from available containers, use :code:`module load <module>`:
+
+    .. code:: bash
+
```

Comment:
add a note that if some modules are missing you have to find the slurm cluster manager to prepare them.

Also, add a note about how to check if a module exists.